File: TLBB1.5/tabletools/CodeTable/CSharp/Table_Card.cs
//This code create by CodeEngine mrd.cyou.com ,don't modify
using System;
 using System.Collections.Generic;
 using System.Collections;

namespace GCGame.Table{

[Serializable]
 public class Tab_Card : ITableOperate
{ private const string TAB_FILE_DATA = "card.txt";
 public enum _ID
 {
 INVLAID_INDEX=-1,
ID_NOTE,
ID_APPEARANCE,
ID_STAR,
ID_NEXT_CARD,
ID_LEVEL_BASE,
ID_MAX_LEVEL,
ID_EXP_BASE,
ID_PRICE_BASE,
ID_SELL_BASE,
ID_LEADER_BASE,
ID_ELEMENT,
ID_ATTACK_TYPE,
ID_ATTACK_BASE,
ID_ATTACK_GROW,
ID_HP_BASE,
ID_HP_GROW,
ID_STORM_BASE,
ID_TENACITY_BASE,
ID_PRECISE_BASE,
ID_DODGE_BASE,
ID_STORM_RATIO,
ID_ATTACK_STORM,
ID_PH_DF_BASE,
ID_PH_DF_GROW,
ID_MG_DF_BASE,
ID_MG_DF_GROW,
ID_PH_NT_BASE,
ID_MG_NT_BASE,
ID_SKILL_LEADER,
ID_SKILL_COMM,
ID_SKILL_VOL,
ID_PROBABILITY,
ID_SKILL_COMB,
ID_COMB_GROUP,
ID_IS_BOSS,
ID_DROP_ID,
ID_HIGHSTAR_DISPLAY,
ID_DROP_LEVEL,
ID_GAMBLE_LEVEL,
ID_CARD_TYPE,
ID_SKILL_STUDY,
ID_SKILL_STUDYLV,
MAX_RECORD
 }
 public string GetInstanceFile(){return TAB_FILE_DATA; }

private int m_Appearance;
 public int Appearance { get{ return m_Appearance;}}

private int m_AttackBase;
 public int AttackBase { get{ return m_AttackBase;}}

private int m_AttackGrow;
 public int AttackGrow { get{ return m_AttackGrow;}}

private int m_AttackStorm;
 public int AttackStorm { get{ return m_AttackStorm;}}

private int m_AttackType;
 public int AttackType { get{ return m_AttackType;}}

private int m_CardType;
 public int CardType { get{ return m_CardType;}}

private int m_CombGroup;
 public int CombGroup { get{ return m_CombGroup;}}

private int m_DodgeBase;
 public int DodgeBase { get{ return m_DodgeBase;}}

private int m_DropID;
 public int DropID { get{ return m_DropID;}}

private int m_DropLevel;
 public int DropLevel { get{ return m_DropLevel;}}

private int m_Element;
 public int Element { get{ return m_Element;}}

private int m_ExpBase;
 public int ExpBase { get{ return m_ExpBase;}}

private int m_GambleLevel;
 public int GambleLevel { get{ return m_GambleLevel;}}

private int m_HighStarDisplay;
 public int HighStarDisplay { get{ return m_HighStarDisplay;}}

private int m_HpBase;
 public int HpBase { get{ return m_HpBase;}}

private int m_HpGrow;
 public int HpGrow { get{ return m_HpGrow;}}

private int m_IsBoss;
 public int IsBoss { get{ return m_IsBoss;}}

private int m_LeaderBase;
 public int LeaderBase { get{ return m_LeaderBase;}}

private int m_LevelBase;
 public int LevelBase { get{ return m_LevelBase;}}

private int m_MaxLevel;
 public int MaxLevel { get{ return m_MaxLevel;}}

private int m_MgDfBase;
 public int MgDfBase { get{ return m_MgDfBase;}}

private int m_MgDfGrow;
 public int MgDfGrow { get{ return m_MgDfGrow;}}

private int m_MgNtBase;
 public int MgNtBase { get{ return m_MgNtBase;}}

private int m_NextCard;
 public int NextCard { get{ return m_NextCard;}}

private string m_Note;
 public string Note { get{ return m_Note;}}

private int m_PhDfBase;
 public int PhDfBase { get{ return m_PhDfBase;}}

private int m_PhDfGrow;
 public int PhDfGrow { get{ return m_PhDfGrow;}}

private int m_PhNtBase;
 public int PhNtBase { get{ return m_PhNtBase;}}

private int m_PreciseBase;
 public int PreciseBase { get{ return m_PreciseBase;}}

private int m_PriceBase;
 public int PriceBase { get{ return m_PriceBase;}}

private int m_Probability;
 public int Probability { get{ return m_Probability;}}

private int m_SellBase;
 public int SellBase { get{ return m_SellBase;}}

private int m_SkillComb;
 public int SkillComb { get{ return m_SkillComb;}}

private int m_SkillComm;
 public int SkillComm { get{ return m_SkillComm;}}

private int m_SkillLeader;
 public int SkillLeader { get{ return m_SkillLeader;}}

private int m_SkillStudy;
 public int SkillStudy { get{ return m_SkillStudy;}}

private int m_SkillStudylv;
 public int SkillStudylv { get{ return m_SkillStudylv;}}

private int m_SkillVol;
 public int SkillVol { get{ return m_SkillVol;}}

private int m_Star;
 public int Star { get{ return m_Star;}}

private int m_StormBase;
 public int StormBase { get{ return m_StormBase;}}

private int m_StormRatio;
 public int StormRatio { get{ return m_StormRatio;}}

private int m_TenacityBase;
 public int TenacityBase { get{ return m_TenacityBase;}}

public bool LoadTable(Hashtable _tab)
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),SerializableTable,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }
 public void SerializableTable(ArrayList valuesList,string skey,Hashtable _hash)
 {
 if (string.IsNullOrEmpty(skey))
 {
 throw TableException.ErrorReader("Read File{0} as key is Empty Fail!!!", GetInstanceFile());
 }

 if ((int)_ID.MAX_RECORD!=valuesList.Count)
 {
 throw TableException.ErrorReader("Load {0} error as CodeSize:{1} not Equal DataSize:{2}", GetInstanceFile(),_ID.MAX_RECORD,valuesList.Count);
 }
 Int32 nKey = Convert.ToInt32(skey);
 Tab_Card _values = new Tab_Card();
 _values.m_Appearance =  Convert.ToInt32(valuesList[(int)_ID.ID_APPEARANCE] as string);
_values.m_AttackBase =  Convert.ToInt32(valuesList[(int)_ID.ID_ATTACK_BASE] as string);
_values.m_AttackGrow =  Convert.ToInt32(valuesList[(int)_ID.ID_ATTACK_GROW] as string);
_values.m_AttackStorm =  Convert.ToInt32(valuesList[(int)_ID.ID_ATTACK_STORM] as string);
_values.m_AttackType =  Convert.ToInt32(valuesList[(int)_ID.ID_ATTACK_TYPE] as string);
_values.m_CardType =  Convert.ToInt32(valuesList[(int)_ID.ID_CARD_TYPE] as string);
_values.m_CombGroup =  Convert.ToInt32(valuesList[(int)_ID.ID_COMB_GROUP] as string);
_values.m_DodgeBase =  Convert.ToInt32(valuesList[(int)_ID.ID_DODGE_BASE] as string);
_values.m_DropID =  Convert.ToInt32(valuesList[(int)_ID.ID_DROP_ID] as string);
_values.m_DropLevel =  Convert.ToInt32(valuesList[(int)_ID.ID_DROP_LEVEL] as string);
_values.m_Element =  Convert.ToInt32(valuesList[(int)_ID.ID_ELEMENT] as string);
_values.m_ExpBase =  Convert.ToInt32(valuesList[(int)_ID.ID_EXP_BASE] as string);
_values.m_GambleLevel =  Convert.ToInt32(valuesList[(int)_ID.ID_GAMBLE_LEVEL] as string);
_values.m_HighStarDisplay =  Convert.ToInt32(valuesList[(int)_ID.ID_HIGHSTAR_DISPLAY] as string);
_values.m_HpBase =  Convert.ToInt32(valuesList[(int)_ID.ID_HP_BASE] as string);
_values.m_HpGrow =  Convert.ToInt32(valuesList[(int)_ID.ID_HP_GROW] as string);
_values.m_IsBoss =  Convert.ToInt32(valuesList[(int)_ID.ID_IS_BOSS] as string);
_values.m_LeaderBase =  Convert.ToInt32(valuesList[(int)_ID.ID_LEADER_BASE] as string);
_values.m_LevelBase =  Convert.ToInt32(valuesList[(int)_ID.ID_LEVEL_BASE] as string);
_values.m_MaxLevel =  Convert.ToInt32(valuesList[(int)_ID.ID_MAX_LEVEL] as string);
_values.m_MgDfBase =  Convert.ToInt32(valuesList[(int)_ID.ID_MG_DF_BASE] as string);
_values.m_MgDfGrow =  Convert.ToInt32(valuesList[(int)_ID.ID_MG_DF_GROW] as string);
_values.m_MgNtBase =  Convert.ToInt32(valuesList[(int)_ID.ID_MG_NT_BASE] as string);
_values.m_NextCard =  Convert.ToInt32(valuesList[(int)_ID.ID_NEXT_CARD] as string);
_values.m_Note =  valuesList[(int)_ID.ID_NOTE] as string;
_values.m_PhDfBase =  Convert.ToInt32(valuesList[(int)_ID.ID_PH_DF_BASE] as string);
_values.m_PhDfGrow =  Convert.ToInt32(valuesList[(int)_ID.ID_PH_DF_GROW] as string);
_values.m_PhNtBase =  Convert.ToInt32(valuesList[(int)_ID.ID_PH_NT_BASE] as string);
_values.m_PreciseBase =  Convert.ToInt32(valuesList[(int)_ID.ID_PRECISE_BASE] as string);
_values.m_PriceBase =  Convert.ToInt32(valuesList[(int)_ID.ID_PRICE_BASE] as string);
_values.m_Probability =  Convert.ToInt32(valuesList[(int)_ID.ID_PROBABILITY] as string);
_values.m_SellBase =  Convert.ToInt32(valuesList[(int)_ID.ID_SELL_BASE] as string);
_values.m_SkillComb =  Convert.ToInt32(valuesList[(int)_ID.ID_SKILL_COMB] as string);
_values.m_SkillComm =  Convert.ToInt32(valuesList[(int)_ID.ID_SKILL_COMM] as string);
_values.m_SkillLeader =  Convert.ToInt32(valuesList[(int)_ID.ID_SKILL_LEADER] as string);
_values.m_SkillStudy =  Convert.ToInt32(valuesList[(int)_ID.ID_SKILL_STUDY] as string);
_values.m_SkillStudylv =  Convert.ToInt32(valuesList[(int)_ID.ID_SKILL_STUDYLV] as string);
_values.m_SkillVol =  Convert.ToInt32(valuesList[(int)_ID.ID_SKILL_VOL] as string);
_values.m_Star =  Convert.ToInt32(valuesList[(int)_ID.ID_STAR] as string);
_values.m_StormBase =  Convert.ToInt32(valuesList[(int)_ID.ID_STORM_BASE] as string);
_values.m_StormRatio =  Convert.ToInt32(valuesList[(int)_ID.ID_STORM_RATIO] as string);
_values.m_TenacityBase =  Convert.ToInt32(valuesList[(int)_ID.ID_TENACITY_BASE] as string);

 _hash.Add(nKey,_values); }


}
}

File: TLBB1.5/tabletools/CodeTable/Java/xjgame/table/Table_Card.java
//This code create by CodeEngine mrd.cyou.com ,don't modify
package xjgame.table;
import java.util.ArrayList;
 import java.util.HashMap;

public class Table_Card implements ITableOperate {
 public final static String TAB_FILE_DATA = "/card.txt";
 public static final int ID_NOTE = 0 ;
public static final int ID_APPEARANCE = 1 ;
public static final int ID_STAR = 2 ;
public static final int ID_NEXT_CARD = 3 ;
public static final int ID_LEVEL_BASE = 4 ;
public static final int ID_MAX_LEVEL = 5 ;
public static final int ID_EXP_BASE = 6 ;
public static final int ID_PRICE_BASE = 7 ;
public static final int ID_SELL_BASE = 8 ;
public static final int ID_LEADER_BASE = 9 ;
public static final int ID_ELEMENT = 10 ;
public static final int ID_ATTACK_TYPE = 11 ;
public static final int ID_ATTACK_BASE = 12 ;
public static final int ID_ATTACK_GROW = 13 ;
public static final int ID_HP_BASE = 14 ;
public static final int ID_HP_GROW = 15 ;
public static final int ID_STORM_BASE = 16 ;
public static final int ID_TENACITY_BASE = 17 ;
public static final int ID_PRECISE_BASE = 18 ;
public static final int ID_DODGE_BASE = 19 ;
public static final int ID_STORM_RATIO = 20 ;
public static final int ID_ATTACK_STORM = 21 ;
public static final int ID_PH_DF_BASE = 22 ;
public static final int ID_PH_DF_GROW = 23 ;
public static final int ID_MG_DF_BASE = 24 ;
public static final int ID_MG_DF_GROW = 25 ;
public static final int ID_PH_NT_BASE = 26 ;
public static final int ID_MG_NT_BASE = 27 ;
public static final int ID_SKILL_LEADER = 28 ;
public static final int ID_SKILL_COMM = 29 ;
public static final int ID_SKILL_VOL = 30 ;
public static final int ID_PROBABILITY = 31 ;
public static final int ID_SKILL_COMB = 32 ;
public static final int ID_COMB_GROUP = 33 ;
public static final int ID_IS_BOSS = 34 ;
public static final int ID_DROP_ID = 35 ;
public static final int ID_HIGHSTAR_DISPLAY = 36 ;
public static final int ID_DROP_LEVEL = 37 ;
public static final int ID_GAMBLE_LEVEL = 38 ;
public static final int ID_CARD_TYPE = 39 ;
public static final int ID_SKILL_STUDY = 40 ;
public static final int ID_SKILL_STUDYLV = 41 ;
public static final int MAX_RECORD = 42 ;

 public String GetInstanceFile(){return TAB_FILE_DATA; }

private int m_Appearance;
 public int GetAppearance() { return m_Appearance;}

private int m_AttackBase;
 public int GetAttackBase() { return m_AttackBase;}

private int m_AttackGrow;
 public int GetAttackGrow() { return m_AttackGrow;}

private int m_AttackStorm;
 public int GetAttackStorm() { return m_AttackStorm;}

private int m_AttackType;
 public int GetAttackType() { return m_AttackType;}

private int m_CardType;
 public int GetCardType() { return m_CardType;}

private int m_CombGroup;
 public int GetCombGroup() { return m_CombGroup;}

private int m_DodgeBase;
 public int GetDodgeBase() { return m_DodgeBase;}

private int m_DropID;
 public int GetDropID() { return m_DropID;}

private int m_DropLevel;
 public int GetDropLevel() { return m_DropLevel;}

private int m_Element;
 public int GetElement() { return m_Element;}

private int m_ExpBase;
 public int GetExpBase() { return m_ExpBase;}

private int m_GambleLevel;
 public int GetGambleLevel() { return m_GambleLevel;}

private int m_HighStarDisplay;
 public int GetHighStarDisplay() { return m_HighStarDisplay;}

private int m_HpBase;
 public int GetHpBase() { return m_HpBase;}

private int m_HpGrow;
 public int GetHpGrow() { return m_HpGrow;}

private int m_IsBoss;
 public int GetIsBoss() { return m_IsBoss;}

private int m_LeaderBase;
 public int GetLeaderBase() { return m_LeaderBase;}

private int m_LevelBase;
 public int GetLevelBase() { return m_LevelBase;}

private int m_MaxLevel;
 public int GetMaxLevel() { return m_MaxLevel;}

private int m_MgDfBase;
 public int GetMgDfBase() { return m_MgDfBase;}

private int m_MgDfGrow;
 public int GetMgDfGrow() { return m_MgDfGrow;}

private int m_MgNtBase;
 public int GetMgNtBase() { return m_MgNtBase;}

private int m_NextCard;
 public int GetNextCard() { return m_NextCard;}

private String m_Note;
 public String GetNote() { return m_Note;}

private int m_PhDfBase;
 public int GetPhDfBase() { return m_PhDfBase;}

private int m_PhDfGrow;
 public int GetPhDfGrow() { return m_PhDfGrow;}

private int m_PhNtBase;
 public int GetPhNtBase() { return m_PhNtBase;}

private int m_PreciseBase;
 public int GetPreciseBase() { return m_PreciseBase;}

private int m_PriceBase;
 public int GetPriceBase() { return m_PriceBase;}

private int m_Probability;
 public int GetProbability() { return m_Probability;}

private int m_SellBase;
 public int GetSellBase() { return m_SellBase;}

private int m_SkillComb;
 public int GetSkillComb() { return m_SkillComb;}

private int m_SkillComm;
 public int GetSkillComm() { return m_SkillComm;}

private int m_SkillLeader;
 public int GetSkillLeader() { return m_SkillLeader;}

private int m_SkillStudy;
 public int GetSkillStudy() { return m_SkillStudy;}

private int m_SkillStudylv;
 public int GetSkillStudylv() { return m_SkillStudylv;}

private int m_SkillVol;
 public int GetSkillVol() { return m_SkillVol;}

private int m_Star;
 public int GetStar() { return m_Star;}

private int m_StormBase;
 public int GetStormBase() { return m_StormBase;}

private int m_StormRatio;
 public int GetStormRatio() { return m_StormRatio;}

private int m_TenacityBase;
 public int GetTenacityBase() { return m_TenacityBase;}

public boolean LoadTable(HashMap<Integer, Table_Card> _tab) throws TableException
 {
 if(!TableManager.ReaderPList(GetInstanceFile(),this,_tab))
 {
 throw TableException.ErrorReader("Load File{0} Fail!!!",GetInstanceFile());
 }
 return true;
 }

 @Override
 public void SerializableTable(ArrayList<String> valuesList,String skey,Object obj) throws TableException {
 @SuppressWarnings("unchecked")
 HashMap<Integer, Table_Card> _hash = (HashMap<Integer, Table_Card> )obj;
 if (skey.isEmpty())
 {
 throw TableException.ErrorReader("Read File"+GetInstanceFile()+" as key is Empty Fail!!!");
 }

 if (MAX_RECORD!=valuesList.size())
 {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as CodeSize:"+MAX_RECORD+" not Equal DataSize:"+valuesList.size());
 }
 try { 
 Integer nKey = Integer.parseInt(skey);
 Table_Card _values = new Table_Card();
 _values.m_Appearance =  Integer.parseInt((String)valuesList.get(ID_APPEARANCE));
_values.m_AttackBase =  Integer.parseInt((String)valuesList.get(ID_ATTACK_BASE));
_values.m_AttackGrow =  Integer.parseInt((String)valuesList.get(ID_ATTACK_GROW));
_values.m_AttackStorm =  Integer.parseInt((String)valuesList.get(ID_ATTACK_STORM));
_values.m_AttackType =  Integer.parseInt((String)valuesList.get(ID_ATTACK_TYPE));
_values.m_CardType =  Integer.parseInt((String)valuesList.get(ID_CARD_TYPE));
_values.m_CombGroup =  Integer.parseInt((String)valuesList.get(ID_COMB_GROUP));
_values.m_DodgeBase =  Integer.parseInt((String)valuesList.get(ID_DODGE_BASE));
_values.m_DropID =  Integer.parseInt((String)valuesList.get(ID_DROP_ID));
_values.m_DropLevel =  Integer.parseInt((String)valuesList.get(ID_DROP_LEVEL));
_values.m_Element =  Integer.parseInt((String)valuesList.get(ID_ELEMENT));
_values.m_ExpBase =  Integer.parseInt((String)valuesList.get(ID_EXP_BASE));
_values.m_GambleLevel =  Integer.parseInt((String)valuesList.get(ID_GAMBLE_LEVEL));
_values.m_HighStarDisplay =  Integer.parseInt((String)valuesList.get(ID_HIGHSTAR_DISPLAY));
_values.m_HpBase =  Integer.parseInt((String)valuesList.get(ID_HP_BASE));
_values.m_HpGrow =  Integer.parseInt((String)valuesList.get(ID_HP_GROW));
_values.m_IsBoss =  Integer.parseInt((String)valuesList.get(ID_IS_BOSS));
_values.m_LeaderBase =  Integer.parseInt((String)valuesList.get(ID_LEADER_BASE));
_values.m_LevelBase =  Integer.parseInt((String)valuesList.get(ID_LEVEL_BASE));
_values.m_MaxLevel =  Integer.parseInt((String)valuesList.get(ID_MAX_LEVEL));
_values.m_MgDfBase =  Integer.parseInt((String)valuesList.get(ID_MG_DF_BASE));
_values.m_MgDfGrow =  Integer.parseInt((String)valuesList.get(ID_MG_DF_GROW));
_values.m_MgNtBase =  Integer.parseInt((String)valuesList.get(ID_MG_NT_BASE));
_values.m_NextCard =  Integer.parseInt((String)valuesList.get(ID_NEXT_CARD));
_values.m_Note =  (String)valuesList.get(ID_NOTE);
_values.m_PhDfBase =  Integer.parseInt((String)valuesList.get(ID_PH_DF_BASE));
_values.m_PhDfGrow =  Integer.parseInt((String)valuesList.get(ID_PH_DF_GROW));
_values.m_PhNtBase =  Integer.parseInt((String)valuesList.get(ID_PH_NT_BASE));
_values.m_PreciseBase =  Integer.parseInt((String)valuesList.get(ID_PRECISE_BASE));
_values.m_PriceBase =  Integer.parseInt((String)valuesList.get(ID_PRICE_BASE));
_values.m_Probability =  Integer.parseInt((String)valuesList.get(ID_PROBABILITY));
_values.m_SellBase =  Integer.parseInt((String)valuesList.get(ID_SELL_BASE));
_values.m_SkillComb =  Integer.parseInt((String)valuesList.get(ID_SKILL_COMB));
_values.m_SkillComm =  Integer.parseInt((String)valuesList.get(ID_SKILL_COMM));
_values.m_SkillLeader =  Integer.parseInt((String)valuesList.get(ID_SKILL_LEADER));
_values.m_SkillStudy =  Integer.parseInt((String)valuesList.get(ID_SKILL_STUDY));
_values.m_SkillStudylv =  Integer.parseInt((String)valuesList.get(ID_SKILL_STUDYLV));
_values.m_SkillVol =  Integer.parseInt((String)valuesList.get(ID_SKILL_VOL));
_values.m_Star =  Integer.parseInt((String)valuesList.get(ID_STAR));
_values.m_StormBase =  Integer.parseInt((String)valuesList.get(ID_STORM_BASE));
_values.m_StormRatio =  Integer.parseInt((String)valuesList.get(ID_STORM_RATIO));
_values.m_TenacityBase =  Integer.parseInt((String)valuesList.get(ID_TENACITY_BASE));

 _hash.put(nKey,_values);
 } catch (NumberFormatException e) {
 throw TableException.ErrorReader("Load "+GetInstanceFile()+" error as :" + e.getMessage());
 }
 }


}

